tstest/integration/vms/vms_test.go

```diff
@@ -874,6 +889,56 @@ func TestDeriveBindhost(t *testing.T) {
 	t.Log(deriveBindhost(t))
 }
 
+func makeTestNode(t *testing.T, bins *integration.Binaries, controlURL string) {
+	dir := t.TempDir()
+	cmd := exec.Command(
+		bins.Daemon,
+		"--tun=userspace-networking",
+		"--state="+filepath.Join(dir, "state.json"),
+		"--socket="+filepath.Join(dir, "sock"),
+		"--socks5-server=localhost:0",
+	)
+
+	cmd.Env = append(os.Environ(), "NOTIFY_SOCKET="+filepath.Join(dir, "notify_socket"))
+
+	err := cmd.Start()
+	if err != nil {
+		t.Fatalf("can't start tailscaled: %v", err)
+	}
+
+	t.Cleanup(func() {
```

Comment:
nit: `t.Cleanup(cmd.Process.Kill)`

Reply:
oh nevermind, Kill probably returns an error.

Reply:
I can't do that, Kill returns an error because the kernel can deny a kill for permissions reasons. In this case I am assuming that this will work because this is a child process and the tests aren't running with the kernel capability needed to change user ids.